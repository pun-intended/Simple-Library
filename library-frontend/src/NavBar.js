import React, { useContext } from "react";
import "./NavBar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import UserContext from "./UserContext";

const NavBar = () =>{
    const currentUser = useContext(UserContext);
  
  return (
    <div>
      <Navbar expand="md">
        <NavLink exact="true" to="/" className="navbar-brand">
          Library
        </NavLink>
        {currentUser && <NavLink to="/profile" className="navbar-brand">
          {`${currentUser.id} - ${currentUser.first_name}`}
        </NavLink>}

        <Nav className="ml-auto" navbar>
          {currentUser &&
          <NavItem>
            <NavLink to="/books">Books</NavLink>
          </NavItem>}
          {currentUser &&
          <NavItem>
            <NavLink to="/students">Students</NavLink>
          </NavItem>}
          {!currentUser &&
          <NavItem>
            <NavLink to="/login">Login</NavLink>
          </NavItem>}
          {/* {!currentUser &&
          <NavItem>
            <NavLink to="/signup">Signup</NavLink>
          </NavItem>} */}
          {currentUser && 
          <NavItem>
            <NavLink to="/logout"> Logout </NavLink>
          </NavItem>}
        </Nav>
        
      </Navbar>
    </div>
  );
}

export default NavBar;
