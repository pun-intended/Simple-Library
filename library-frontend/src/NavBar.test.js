import React from "react";
import { render } from "@testing-library/react";
import NavBar from "./NavBar";
import {MemoryRouter} from 'react-router-dom'

// Smoke test
it("renders without crashing", () => {
    render(<MemoryRouter> <NavBar /> </MemoryRouter>)
})

// Snapshot
it("matches snapshot", () => {
    const {asFragment} = render(<MemoryRouter> <NavBar /> </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});

// Wrap in context provider
// Displays correctly when logged in

// displays correctly when not logged in