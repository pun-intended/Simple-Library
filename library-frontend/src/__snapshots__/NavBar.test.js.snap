// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
   
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          exact="true"
          href="/"
        >
          Library
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/login"
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
   
</DocumentFragment>
`;
